src/ql/arch/cc/info.cc

```diff
@@ -587,7 +720,22 @@ void Info::post_process_platform(
  */
 void Info::populate_backend_passes(pmgr::Manager &manager, const utils::Str &variant) const {
 
-    // Mimic original CC backend.
+    // Remove passes we don't need/want.
+    // FIXME: bit of a hack, and invalidates https://openql.readthedocs.io/en/latest/gen/reference_architectures.html#default-pass-list
+    const auto &prescheduler = com::options::global["prescheduler"];
+    if (prescheduler.as_bool() && prescheduler.get_default() == prescheduler.as_str()) {    // prescheduler enabled implicitly
+        const utils::Str passes[] = {
+            "prescheduler",         // enabled by default, but pointless since we add our own scheduling
+            "scheduledqasmwriter"   // confusing, since it doesn't use our scheduling
```

Comment:
The scheduledqasmwriter pass should always be removed, since you always add it later in a later place. See also next comment.